File: GUI/src/app/category.ts
export class Category{
    id: Number;
    name: string;
    isPublic: boolean;
    createdBy:string;
    creationDate:Date;
  
    constructor(name:string='', isPublic:boolean=false, createdBy:string='', createdDate:Date=null) {
      this.name = name;
      this.isPublic = isPublic;    
      this.createdBy = createdBy;
      this.creationDate = createdDate;
    }    
}  

File: CategoryService/CategoryService.API/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CategoryService.API.Models
{
    public class Category
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool IsPublic { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreationDate { get; set; }
    }
}
